src/disco/containers/DiscoPane.js

```diff
@@ -67,10 +67,19 @@ export class DiscoPaneBase extends React.Component {
     // instead of an empty list because an empty list could be a valid
     // (yet unlikley) API response.
     if (!errorHandler.hasError() && !results.length) {
+      // We accept all query params here and filter them out based on the
+      // `taarParamsToUse` config value. See:
+      // https://github.com/mozilla/addons-frontend/issues/4155
+      const taarParams = { platform: params.platform, ...location.query };
+      // We translate `clientId` to `'telemetry-client-id'`.
+      if (taarParams.clientId) {
```

Comment:
I wonder if this would be better moved to where the filtering happens?  Also it's worth noting this implicitly allows telemetry-client-id as a query param too, not that this probably matters.

Also it looks like `platform` from the URL could be clobbered by `platform` as a query param. You'd need to reverse the order to prevent that. Would be good to add a test for that case.